extensions/hibernate-orm/deployment/src/main/java/io/quarkus/hibernate/orm/deployment/HibernateEntityEnhancer.java

```diff
@@ -43,30 +44,14 @@ private static class HibernateEnhancingClassVisitor extends QuarkusClassVisitor
         private final ClassVisitor outputClassVisitor;
         private final Enhancer enhancer;
 
-        public HibernateEnhancingClassVisitor(String className, ClassVisitor outputClassVisitor) {
+        public HibernateEnhancingClassVisitor(String className, ClassVisitor outputClassVisitor,
+                Enhancer enhancer) {
             //Careful: the ASM API version needs to match the ASM version of Gizmo, not the one from Byte Buddy.
             //Most often these match - but occasionally they will diverge which is acceptable as Byte Buddy is shading ASM.
             super(Gizmo.ASM_API_VERSION, new QuarkusClassWriter(ClassWriter.COMPUTE_FRAMES | ClassWriter.COMPUTE_MAXS));
             this.className = className;
             this.outputClassVisitor = outputClassVisitor;
-            //note that as getLoadingClassLoader is resolved immediately this can't be created until transform time
-
-            DefaultEnhancementContext enhancementContext = new DefaultEnhancementContext() {
-
-                @Override
-                public boolean doBiDirectionalAssociationManagement(final UnloadedField field) {
-                    //Don't enable automatic association management as it's often too surprising.
-                    //Also, there's several cases in which its semantics are of unspecified,
-                    //such as what should happen when dealing with ordered collections.
-                    return false;
-                }
-
-                @Override
-                public ClassLoader getLoadingClassLoader() {
-                    return Thread.currentThread().getContextClassLoader();
-                }
-            };
-            this.enhancer = PROVIDER.getEnhancer(enhancementContext);
+            this.enhancer = enhancer;
```

Comment:
This is the crux of the patch: reuse a single `Enhancer` instace for all processed entities.

Creating an enhancer is extremely taxing, especially on the classloader as ByteBuddy needs to load a gazillion of bytecode streams to generate its teamplates - we designed it intentionally so to preapre all the possible templates in advance to then process multiple entities efficiently.

Reply:
Does this mean the enhancer is shared across multiple persistence units? Is that safe?

Reply:
It should be yes - the entity enhancement does not depend on the PU.

Also this way of processing things is more similar to how ORM's build utilities would do it: the Enhancer was meant to be reused on a batch of files.